account_mtd/models/mtd_hello_world.py

```diff
@@ -0,0 +1,456 @@
+# -*- coding: utf-8 -*-
+
+import requests
+import json
+import logging
+import werkzeug
+import urllib
+
+from odoo import models, fields, api, exceptions
+from datetime import datetime, timedelta
+from urllib.parse import urlparse
+
+
+class MtdHelloWorld(models.Model):
+    _name = 'mtd.hello_world'
+    _description = "Hello world to test connection between Odoo and HMRC"
+    
+    _logger = logging.getLogger(__name__)
+    
+    name = fields.Char('Name', required=True, readonly=True)
+    api_name = fields.Many2one(comodel_name="mtd.api", required=True)
+    hmrc_configuration = fields.Many2one(comodel_name="mtd.hmrc_configuration", string='HMRC Configuration')
+    scope = fields.Char(related="api_name.scope")
+    response_from_hmrc = fields.Text(string="Response From HMRC", readonly=True)
+    which_button_type_clicked = fields.Char(string="which_button")
+    path = fields.Char(string="sandbox_url")
+    current_record = fields.Char()
+    # need a variable which will keep the record id so that when we
+    
+    @api.multi
+    def action_hello_world_connection(self):
+        if not self.hmrc_configuration:
+            raise exceptions.Warning("Please select HMRC configuration before continuing!")
+        # making sure that the endpoint is what we have used in the data file as the name can be changed anytime.
+        endpoint_record = self.env['ir.model.data'].search([
+            ('res_id', '=', self.id), ('model', '=', 'mtd.hello_world')
+        ])
+
+        if endpoint_record.name == "mtd_hello_world_endpoint":
+            return self._handle_mtd_hello_world_endpoint()
+        elif endpoint_record.name == "mtd_hello_application_endpoint":
+            return self._handle_mtd_hello_application_endpoint()
+        elif endpoint_record.name == "mtd_hello_user_endpoint":
+            return self._handle_mtd_hello_user_endpoint()
+        else:
+            raise exceptions.Warning("Could not connect to HMRC! \nThis is not a valid HMRC service connection")
+
+    def _handle_mtd_hello_world_endpoint(self):
+        self.which_button_type_clicked = "helloworld"
+        self.path = "/hello/world"
+        version = self._json_command('version')
+        self._logger.info(
```

Comment:
Logged this message in three places, and used it as a prefix in even more.  How about:

```
_logger.info(self._connection_button_clicked_log_message())
```

Where you've appended extra, you can include the result of that call in your format() calls or simply use `+` to concatenate your extra stuff.